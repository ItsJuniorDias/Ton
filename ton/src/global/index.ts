export * from './shadowStyles';
export * from './theme';
