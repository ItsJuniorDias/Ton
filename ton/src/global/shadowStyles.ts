export const shadowStyles = {
  shadowColor: '#000',
  shadowOffset: {
    width: 0,
    height: 5,
  },
  shadowOpacity: 0.1,
  shadowRadius: 2.62,
  elevation: 2,
};
