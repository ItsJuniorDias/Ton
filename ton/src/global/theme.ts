export const theme = {
  colors: {
    primary: '#01ad0b',
    secondary: '#01ea34',
    shape: '#ffffff',
    background: '#eff4f8',
    title: '#000000',
    text: '#20252b',
    notification: '#ff7b00',
    alert: '#ed5e69',
  },
  fonts: {
    regular: 'Roboto-Regular',
    medium: 'Roboto-Medium',
    bold: 'Roboto-Bold',
  },
};
