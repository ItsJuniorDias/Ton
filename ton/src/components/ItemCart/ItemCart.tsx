import React, { useRef } from 'react';
import { Pressable } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { theme } from '../../global';
import { useAppDispatch } from '../../hooks';
import { deleteProductFromCart } from '../../store/modules/cart';

import { CartProduct, Thumbnail, TitleProduct, ContentImage } from './styles';

type ItemCartProps = {
  id: string;
  title: string;
  thumbnail: string;
};

export const ItemCart = ({ id, title, thumbnail }: ItemCartProps) => {
  const dispatch = useAppDispatch();

  const testIDs = useRef({
    pressable: 'pressableItemCart_testID',
  }).current;

  return (
    <CartProduct>
      <ContentImage>
        <Thumbnail source={{ uri: thumbnail }} resizeMode="contain" />

        <TitleProduct>{title}</TitleProduct>
      </ContentImage>

      <Pressable
        testID={testIDs.pressable}
        onPress={() =>
          dispatch(
            deleteProductFromCart({
              id,
            })
          )
        }
        style={({ pressed }) => [
          {
            opacity: pressed ? 0.6 : 1,
          },
        ]}
      >
        <Icon name="highlight-remove" size={28} color={theme.colors.alert} />
      </Pressable>
    </CartProduct>
  );
};
