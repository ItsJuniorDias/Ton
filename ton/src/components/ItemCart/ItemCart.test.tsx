import 'jest-styled-components';
import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { ItemCart } from './ItemCart';
import { theme } from '../../global';
import store from '../../store';
import { Provider } from 'react-redux';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');
jest.mock('react-native-iphone-x-helper', () => ({
  getStatusBarHeight: jest.fn(),
  isIphoneX: jest.fn(),
}));

describe('Behavior ItemCart', () => {
  const props = {
    id: 'T1',
    title: 'T1 Brother',
    thumbnail:
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t1-1',
  };

  const screenRender = () => (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <ItemCart {...props} />
      </ThemeProvider>
    </Provider>
  );

  it('render snapshot', () => {
    const result = render(screenRender()).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('should call function onPress button pressable', () => {
    const { getByTestId } = render(screenRender());

    const pressable = getByTestId('pressableItemCart_testID');

    fireEvent.press(pressable);

    expect(pressable).toBeTruthy();
  });
});
