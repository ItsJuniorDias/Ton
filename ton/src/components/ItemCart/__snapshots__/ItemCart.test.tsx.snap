// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Behavior ItemCart render snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 31,
        "flexDirection": "row",
        "height": 188,
        "justifyContent": "space-between",
        "marginTop": 63,
        "paddingBottom": 31,
        "paddingLeft": 31,
        "paddingRight": 31,
        "paddingTop": 31,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={
        {
          "uri": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t1-1",
        }
      }
      style={
        [
          {
            "height": 126,
            "width": 126,
          },
        ]
      }
    />
    <Text
      style={
        [
          {
            "color": "#20252b",
            "fontFamily": "Roboto-Medium",
            "fontSize": 31,
          },
        ]
      }
    >
      T1 Brother
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
      ]
    }
    testID="pressableItemCart_testID"
  >
    <Icon
      color="#ed5e69"
      name="highlight-remove"
      size={28}
    />
  </View>
</View>
`;
